digiroad2-api-viite/src/main/scala/fi/liikennevirasto/digiroad2/SearchApi.scala

```diff
@@ -70,36 +109,53 @@ class SearchApi(roadAddressService: RoadAddressService,
       tags "SearchAPI (Digiroad)"
       summary "Returns all the existing road numbers at the current Viite road network."
       description "Returns List of all the existing road numbers at the current Viite road network." +
-              "The Viite current network may contain roadway number changes that will be in effect only in the future."
+              "The Viite current network may contain roadway number changes that will be in effect only in the future.\n"+
+              "<br />2023-02-23 The list is now sorted. Better exception handling, and more explicit help for the caller."
       parameter headerParam[String]("X-API-Key").required.description(XApiKeyDescription)
     )
-
   // TODO: "?" in the end is useless; does not take query params
   get("/road_numbers?", operation(getRoadNumbers)) {
-    time(logger, "GET request for /road_numbers?") {
-      roadAddressService.getRoadNumbers
-    }
-  }
+    val requestString = s"GET request for ${request.getRequestURI}  (${getRoadNumbers.operationId})"
+    time(logger, requestString) {
 
+      try {
+        roadAddressService.getRoadNumbers.toList.sorted
+      }
+      catch {
+        case throwable: Throwable => haltWithHTTP500WithLoggerError(requestString, throwable)
+      }
 
+    }
+  }
 
   private val getRoadAddressWithRoadNumber: SwaggerSupportSyntax.OperationBuilder = (
     apiOperation[List[Map[String, Any]]]("getRoadAddressWithRoadNumber")
       .parameters(
         headerParam[String]("X-API-Key").required.description(XApiKeyDescription),
         pathParam[Long]("road").required.description(roadNumberDescription),
-        queryParam[Long]("tracks").optional.description("(Optional) " + trackNumberFilterDescription + "\nIf omitted, any track is returned.")
+        queryParam[Long]("tracks").optional.description("(Optional) " + trackNumberFilterDescription + "\n" +
+          "You may request multiple tracks at a time, by adding several tracks (e.g. ?tracks=1&tracks=2)\n" +
+          "If omitted, any track is returned.\n" +
+          "Also, any invalid 'tracks' entry is omitted. So with invalid only 'tracks' parameters you get unfiltered tracks.")
```

Comment:
"Any invalid 'tracks' entry is omitted", but code seems to throw an exception at any case where 'tracks' input sequence is not a subset of [0,1,2]

Reply:
True! Thanks! The comment still reflected the old implementation, where there were no range checks. Now fixed.